CoEpi/ui/alerts/AlertNotificationsShower.swift

```diff
@@ -0,0 +1,53 @@
+import Foundation
+import UserNotifications
+import UIKit
+import RxSwift
+import os.log
+
+class AlertNotificationsShower {
+    
+    private let disposeBag = DisposeBag()
+
+    // TODO background task / fetch - model flow differently
+    // TODO clear badge
+    // TODO open alerts screen when click on notification
+
+    init(alertsRepo: AlertRepo) {
+        alertsRepo.alerts
+            .distinctUntilChanged()
+            .pairwise()
+            .filter { previous, current in current.count > previous.count }
+            .map { previous, current in current }
+            .subscribe(onNext: { [weak self] alerts in
+                self?.handleNewAlerts(alerts)
+            }).disposed(by: disposeBag)
+    }
+
+    private func handleNewAlerts(_ alerts: [Alert]) {
+        updateAppBadge(number: alerts.count)
+        showNewAlertsNotification()
+    }
+
+    private func updateAppBadge(number: Int) {
+        UNUserNotificationCenter.current().requestAuthorization(options: [.alert, .badge, .sound]) { (granted, error) in
+            if error == nil {
+                DispatchQueue.main.async {
+                    os_log("Updating app badge: %@", log: servicesLog, type: .debug, "\(number)")
+                    UIApplication.shared.applicationIconBadgeNumber = number
+                }
+            }
+        }
+    }
+
+    private func showNewAlertsNotification() {
```

Comment:
@i-schuetz We need to check authorization status before scheduling a notification:
`let center = UNUserNotificationCenter.current()
center.getNotificationSettings { settings in
    guard (settings.authorizationStatus == .authorized) ||
          (settings.authorizationStatus == .provisional) else { return }
    if settings.alertSetting == .enabled {
        // Schedule an alert-only notification.
    } else {
        // Schedule a notification with a badge and sound.
    }
}`

https://developer.apple.com/documentation/usernotifications/asking_permission_to_use_notifications

Reply:
It seems there was a miscommunication issue. This PR is still in progress. @astewart43-gtri Please open it as a draft next time. Not sure also if group review was requested yet.

Reply:
I apologize. I will next time.

Reply:
TIL draft pr: https://github.blog/2019-02-14-introducing-draft-pull-requests/

Reply:
authorization is now checked before notification is scheduled.